tests/tests_pytorch/strategies/test_deepspeed_strategy.py

```diff
@@ -655,8 +661,8 @@ def test_deepspeed_multigpu_stage_3(tmpdir):
         enable_progress_bar=False,
         enable_model_summary=False,
     )
-    trainer.fit(model)
     trainer.test(model)
+    trainer.fit(model)
```

Comment:
Maybe run `.test()` before **and** after the `.fit()` execution?

Reply:
the reason I moved it before is that once you call `trainer.test`, the deepspeed engine doesn't carry over the optimizer and it creates an issue when you try to save the checkpoint which is happening in all the cases where I moved it before.

Reply:
@rohitgr7 are you saying that: 
- before, it didn't work both ways
- now it works both ways
?

The expected behavior is: Whenever fit is called, _initialize_deepspeed_train() should get triggered which calls configure_optimizers if optimizer is not specified in config. 
This is regardless of the order of fit/test. 

Is this correct?

Reply:
no.. previously it worked both ways but now during evaluation we are not configuring optimizers so the sequence:
```py
trainer.fit()
trainer.test() <- does not have optimizers so we can't call save method
trainer.save() <- won't work since there are no optimizers configured with deepspeed engine
```

Reply:
@rohitgr7  Thanks, I was not aware of this. So to be clear, this PR fixes one thing but breaks something else. Would you mind opening a GitHub issue about this newly introduced problem and ping me on it? I'm sure we can fix it by modifying the strategy. 

Not requesting that you work on it, but just to keep track of it. Thanks!

Reply:
done: #15247